pkg/admission/apibinding/apibinding_admission.go

```diff
@@ -208,15 +211,19 @@ func (o *apiBindingAdmission) checkAPIExportAccess(ctx context.Context, user use
 
 // ValidateInitialization ensures the required injected fields are set.
 func (o *apiBindingAdmission) ValidateInitialization() error {
+	if o.initErr != nil {
+		return fmt.Errorf("initialization failed: %w", o.initErr)
+	}
+
 	if o.kubeClusterClient == nil {
 		return fmt.Errorf(PluginName + " plugin needs a Kubernetes ClusterInterface")
 	}
 
 	return nil
 }
 
-// SetKubeClusterClient is an admission plugin initializer function that injects a Kubernetes cluster client into
+// SetClientConfig is an admission plugin initializer function that injects a client config into
 // this admission plugin.
-func (o *apiBindingAdmission) SetKubeClusterClient(clusterClient kubernetes.ClusterInterface) {
-	o.kubeClusterClient = clusterClient
+func (o *apiBindingAdmission) SetClientConfig(cfg *rest.Config) {
+	o.kubeClusterClient, o.initErr = kubernetes.NewClusterForConfig(authorization.WithDeepSARConfig(rest.CopyConfig(cfg)))
```

Comment:
would prefer to have a separate client for SAR, `o.deepSARClusterClient` or so

Reply:
makes sense, plus it gets rid of the ugly init err handling if we pass a pre-constructed client.

i wasn't sure if injecting a general config might be useful in the future.